### ISSUE TITLE
Invalid parameter expression not reported and causes incorrect display of signatures.

### ISSUE THREAD

**Issue by Daraan** [microsoft/pyright#9088]
**Describe the bug**
Recently I've written quite a few tests for `TypeAliasType`, these revealed some inconsistent signatures in pyright (labled C, D, E below) and I think some missing errors where Concatenate should be used for parameter expressions (B). [PEP 612](https://peps.python.org/pep-0612/#user-defined-generic-classes) mostly discusses the rules for `Generics` but I think they should be applied here as well.

I think not using a correct parameter expression causes the C-E cases. It's then a design decision if they should even be fixed as the root cause is the incorrect usage B. If C-E stay as they are they at least hint that something went wrong before.

**Code or Screenshots**

```python
from typing import Callable, Concatenate, ParamSpec, Any, TypeVar, Unpack, Tuple
from typing_extensions import TypeAliasType, TypeVarTuple

T = TypeVar("T")
P = ParamSpec("P")
Ts = TypeVarTuple("Ts")

# Use single ParamSpec
CallableP = TypeAliasType("CallableP", Callable[P, Any], type_params=(P, ))

ImplicitConcatP = CallableP[[int, P]]            # <--(B.1.1) Invalid parameters_expression, should use Concatenate here
invalid_concat_usage = ImplicitConcatP[str]  # <--(C.1) expected signature (int, str); not (int, (str

CorrectConcatP = CallableP[Concatenate[int, P]]  # <--(D.1) expected signature "(int, **P)"; not "(Concatenate[int, P@callableP2])"
concat_usage = CorrectConcatP[str]    # <--(E.1) expected signature (int, str); not "(Concatenate[int, (str)]) -> Any"

### Use ParamSpec and T
Callable2PT = TypeAliasType("Callable2PT", Callable[P, T], type_params=(P, T))
ImplicitConcatPT = Callable2PT[[int, P], object]  # <--(B.2.1) invalid parameters_expression should use Concatenate
invalid_concat_usage2 = ImplicitConcatPT[str] # <--(C.2) expected signature (int, str) not "(int, (str)) -> object"

# This is correct
CorrectConcatPT = Callable2PT[Concatenate[int, P], object]  # D.2 OK: (int, **P@CorrectConcatPT) -> object
concat_usage2 = CorrectConcatPT[str]              # E.2 OK (int, str) -> object
```

Minimal code sample in [pyright playground](https://pyright-play.net/?strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDSFgoDGRM1KrdUACkSERADKCak3oBBFHHoAVRNQBqfAFChIsRKjQB9agA82KAM5JGRzNlz45I8SSREjNrs6UgZAVwQ1lymVABeKFc%2BAAoAIhlwgEplbkCePgFhUQjuGL9zIJD3LxoImSMM5QBiKABVI2ooE3QaRP4hESZlYjJKGnjs%2BTsHJ3kItvIqanT6IY7qAG1uCSkAXXp4EW0EJIgjANDZqGjY5QBJbHsmJBgCRhYYLsJSYc6pqdQYem55%2BahPr%2B-PsoAeAC0ANCACEAHQARkh0SgBxQADdSEgACZQNaNahsEBGXR6BAgahGEyMehGAAWYA8JFRHiqDGYnA4bCgZOoBOUqER9mR2iYl1Y2lpRDQ1SCR28SFO5351ymRhgIA%2BUH%2BQNCBGhUH0zTYqJMaCZHgJUFCz1JCuiAG4oCgwPgTSgXsb5SBfBcQASmNKGdcEhMRtwphdvexOE8Ha93r8oIDgQARDVa0Q6mpIfWsQ3VcL2x0AKhz3Gi4StNvwWaDVxDbDDju4AAEWO1-QAmeaF5R872CozC0UMd1J8usAPOpVRmOhACiCbxSeoutTBqN2bNIEt1ttUDLMsr01NTvNragAIAfFBJHBwr4SteKnTeI0UkwoEQUKiZK07pMm9x-N1bPZHGcQZPxGb8onGECaBmWRFg0FZ0QETZtlkPZDmOSUzkHa5f1uRsaDAx49zeegwAoAArJMlRVYFwSbDVOSRVEEIgTE2RxLUCSJUwUBqCkqRpOksJ3DkEUY3kZS7HsmwScUTkwmUfzlBUPmotUwSbGFE09OcUzTGAM2NPdnRhEtN2XfdVxhE8oFIijPUvUpgjJJBzBcqA%2BX7T1WlwD0vSuH9fUg6gCKEpld3DHhYNsyio3jaSAHkAGkAC5DIivM6zdXysJ-KzT2iryOyuSSRWkoIsoHBSZCUxUfjqr4yinBLErSx1jKPU9lCAA)

This [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDSFgoDGRM1KrdUACkSERADKCak3oBBFHHoAVRNQBqfegFUUCIkwDWsgK4IaAKFCRYiVGgD61AB5sUAZySMHmbLnxyR4kkiIOvLkClEBl9I0MZKABeKGC%2BAAoAIhkkgEpDbhiePgFhUWTudMjXWPjQ8OpkmQdiwwBiRqbmlta2hsJScipqOPkoHz9XQIa2sfGmww7BCxoc-mh8pkNiMkoaLLL5Qf9A5NXujaT6A-XqAG1uCSkAXXp4EUsNBYdohKuoNIykn4B5FBIcCgRHmeRETDMIigzwEPySU3qUAAtCjUVAlL4ACadBzUVyo1EdBKoGBpZEAPgGUkMLDWPUsJOypx63HOJJuCKgxJQMHoJLJSMpkjgNK6Z0sACYSZZfA4YOImWKWWyeXyeRyOgSUVAAPw8cA9CCsJAQ4i41x6rVIwyYsC6emY6jAVDUbGxZkbFUwG5QKCIgA8KISAC0yQARO09KBhp2oJAwZwoKBMAAWRHQeNgYDAJATCCzUCSAHF2NQQCadrUoCgwPgktzSRSqXAksCpBBcNQbZGaJZHc6UK7FXTPezfQGg6GoCBqDg8FAHJBqDAUxYdZyGwKhVIoAB6LkktWNwXN0Uj6gMnkypBy4eHaisr03Dm0%2B%2BS6WymAAITvZ0fh8wdUNURBIZC3U9XzOKJ3SVT0ZA5REZFXVxqAAR10JAADdSHYfAWFxKBbUzGs8JnThgSgbCSF0XoV1YLkmF0EAZx5QEyU0JhZxgVwOlxXoUxgGAEAcAAuXddzQeMU3tAA6JhIF3BA4BXRhdyYJSVJQXcbwcGiHF3ABGCUABZTIAVjPe8ZGvW8YPPR94I5GAQDgETDF9X0SMsKikDdZNYOoazPyfQxbE4hBPHkABRZjcGBFC3I86Fyx5ZJbHBNhsUCGLwBAY4oGoDIiX5JthUMHy-MgnoZBC6YUztEhsQoXo0IwqjcMMXSKDleNdDYX9lQAuUQA5LqepgPqL0-AbPS9ehhufBEJlaDoVAIghGBYexOFGZbmg6LADBNeMAG5AIq4F8AAJV0HksGoGTitVHhwLKjbmFYAB1KS7RgAAZG98Ds%2B9WQA7gfSSyHxwXerdEaqBmvO0hfMMABJbBfCYeN3q2zZOns845p4Z8oahickQSL8ZIM6myVQC6YQgZcywcawbAQGcHCcRh5th%2BHdHWzbOA4fqUzLLtnNc9yPPp5HMUseSPpgSwBaINBelidGjqxmAcdYVkFuhwMKYIWmCvZ0RMoXJA0BFpjegbebnLSM6SIPZ6EmGr5QpscLIpEHK4v8ArEo8jmSTSi2mCt7LYry%2BhCs5Nbej17b%2BoxXyoE%2Bog4HXDbmMt1O8Y9B9zlT9hOCJ8GIfJhIwzN9LLaHJxbdYe3C0dqAACou%2B4NIkld2sO-LkWLjBgABKqNglG5%2B8MfxcXnfOZ2jouZMsJ5ciZtgQFZywYlid56DSegkkb6OhyKGkheV1X1aZXAV91m%2BDeciHSbJqBjYSKKG6jq2W52xnO7XkC5naDzrAkEelcAKe2drPUqUh4SajRFdV0YAoAAHcpI8DbFlKYJcJTcGgn0bwvhdjyH2AFIhqQTgBUuLIO4kILyM1eEfOI3s4S8AWEsPBbZaxixAKQxQfA4ScgztiM0eInrHkpCSSyZwJSXiBvje8NDCbsiPMBLkYEmwyAUT0CUMgSGEOIYTeCjCNwlRPPIiqEpshT2oEYmq7IrHPWsZSMAFAABWlsDE0CUVKK8n4FTA0UWYommioBeN8dHbRqMUAXXwr0MAwAQHgRiX4xxSjpS5MSXLGaTizEAUyXE6GDg%2BZNV6LLLEnItRQDQbaaJaSOZeJoEaBMEJkn4gJB0BJsBkL%2BQImmTCtEMEC16NNYOzpd74EZszEAbjZHRJ8Vk6hyibIqNMTVJ89BSnehkUeDJqzo7%2BKce%2BYJgMfxhMMREgCUT9naJundJmUAEkXSkfQCgfVqwYIEWWLBOcCyYPLP1VAAybxDK7NkoKgNCnqPOBYlZsSDmS1Dp5Ws3kCmxBhZsp80NY65R9n7YRgchHB2oOi5KEcz7-yHIS3A%2BVE67T2o0VagslYVzYCy1lB0MbHRgGdGpmd6LPITEzR6WtMbYxfiY6hdznrgz2Sc70RsgxUwlGbYV2J5k71ZulTm3MkwVIatiCZDBOWj0MGi6WvptUKxvirBwasnHZClQKouNVDa11NhKMk58AE2yAQ7IaztfmQNgV7Y5KL4RhS4qSuO8UQ62upalWlGV6XRTjkyoqiJ0rSpgCsR%2BhdZXwrMdAtgVcmGPOhvXexvwADSIkQH0B7twcey8S1K2IdGvxC8yz4E7avWV69N4LAWXvA%2BXIPgn0LAGy%2B8JFZbSdS6%2Bx7pi3Du7V6t%2BH9P6-3rQ2ltYCQC9tOSg60hgQh%2BExCabIgRKx7CSFeogN6mD5TCAYMez01AaG0Iihwz57jyDHQINhNRaDewgEgGwrpLBjSYOWEQmJiHZGfa%2Bsu9CK1ftAcTRhCE4iVCJp3XRJ5HndkHHB%2B0DgENIAirB0JUAoMwflvBxDroIlIprZ-Z5y57pQDJYBHgyl6pJgAMwyQAJz0BYCgN2iNWO0cyuRi8Cm6Py3EJeC6sQmOwdU%2BxmqMhlUop9IiE1cMqlI1qf2%2BctoKOqcypYUJsRbMqaozRtTjnNMFI8h0VqpBBPcGE4wKA4mDIAAYADUhggA) is a more comprehensive version that also contains correct cases aside from the ones reported above. Locations of interest are marked with `<--`.

**VS Code extension or command-line**
Playground 1.1.382

**Comment by erictraut** [comment]
Thanks for the bug report and the clear repro steps and code sample. There were several bugs here. They will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.383.
